ICAFIX/scripts/computeRecleanFeatures.m

```diff
@@ -0,0 +1,273 @@
+VisualROI=VisualROI_CIFTI.cdata;
+NonVisualROI=1-VisualROI_CIFTI.cdata;
+
+LanguageROI_CIFTI=ciftiopen(LanguageAreasFile,'wb_command');
+LanguageROI=LanguageROI_CIFTI.cdata;
+NonLanguageROI=1-LanguageROI_CIFTI.cdata;
+
+SubRegionsROI=ciftiopen(SubRegionsFile,'wb_command');
+
+FinalSpatialSmoothingFWHM=4;
+
+% Calculate AdditionalSigma
+%AdditionalSigma = FinalSpatialSmoothingFWHM / (2 * (sqrt(2 * log(2))));
+%AdditionalSigmaString=num2str(AdditionalSigma, '%0.20f');
+
+subj=Subject;
+SubjFolderlist=[StudyFolder '/' subj];
+
+for j=1:length(fMRINames)
+    fMRIName=fMRINames{j};
+
+    sICA=load([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_mix']);
+    num_comps=size(sICA,2);
+
+    OrigFeatures=single(zeros(num_comps,186));
+    NewFeatures=single(zeros(num_comps,424));
+
+    MelodicFile=[SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_mix'];
+    if exist(MelodicFile,'file')
+        sICA=load(MelodicFile);
+        if isfile([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/fix4melview_HCP_Style_Single_Multirun_Dedrift_thr10.txt'])
+            prob_file = importdata([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/fix4melview_HCP_Style_Single_Multirun_Dedrift_thr10.txt']);
+        elseif isfile([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/fix4melview_HCP_hp2000_thr10.txt'])
+            prob_file = importdata([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/fix4melview_HCP_hp2000_thr10.txt']);
+        end
+        probability=prob_file.data(1:size(sICA,2),:);
+        Stats=load([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_ICstats']);
+        FeaturesFile=load([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/fix/features.csv']);
+        Powerspectra=load([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_FTmix']);
+
+        % derive TR
+        file_name=[SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_Atlas.dtseries.nii'];
+        CIFTIDenseTimeSeries=ciftiopen(file_name, "wb_command");
+        TR = CIFTIDenseTimeSeries.diminfo{2}.seriesStep;
+
+        PowerspectraScale=[(1/(TR*2))/size(Powerspectra,1):(1/(TR*2))/size(Powerspectra,1):1/(TR*2)];
+        StandardPowerspectraScale=[(1/(TR*2))/(RunsXNumTimePoints/2):(1/(TR*2))/(RunsXNumTimePoints/2):1/(TR*2)];
+        sICAweighted=sICA.*repmat(Stats(:,1)',length(sICA),1);
+        sICAnoisy=sum(abs(sICA),2);
+        sICAnoisy=sICAnoisy>prctile(sICAnoisy,87.5);
+
+        CIFTI=ciftiopen([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_oIC.dscalar.nii'],'wb_command');
```

Comment:
For future reference, `CIFTI` is not a good name for a variable, it doesn't tell you what the data values represent.